src/components/Card.tsx

```diff
@@ -0,0 +1,356 @@
+        padding: 0;
+    }
+
+    ${breakpoints.desktop`
+        .video-step &, .interactive-step & {
+            .openstax-exercise-badges {
+              margin-right: 3.8rem;
+            }
+        }
+    `}
+
+    ${breakpoints.mobile`
+        .openstax-exercise-badges svg {
+            margin-right: ${breakpoints.margins.mobile.length};
+        }
+    `}
+
+    &&& {
+        .openstax-has-html .splash .frame-wrapper { margin-top: 0; }
+    }
+`;
+LoadingCard.displayName = 'LoadingCard';
+
+interface SharedProps {
+  questionNumber: number;
+  numberOfQuestions: number;
+  leftHeaderChildren?: ReactNode;
+  rightHeaderChildren?: ReactNode;
+  headerTitleChildren?: ReactNode;
+}
+
+export interface StepCardProps extends SharedProps {
+  unpadded: boolean;
+  className?: string;
+  children?: ReactNode;
+  stepType: Step['type'];
+  availablePoints: availablePoints;
+  exerciseId?: string;
+  multipartBadge?: ReactNode;
+  isHomework: boolean;
+}
+
+const StepCard = ({
+  questionNumber,
+  numberOfQuestions,
+  stepType,
+  isHomework,
+  availablePoints,
+  unpadded,
+  className,
+  children,
+  exerciseId,
+  multipartBadge,
+  leftHeaderChildren,
+  rightHeaderChildren,
+  headerTitleChildren,
+  ...otherProps }: StepCardProps) => {
+  return (
+    <OuterStepCard {...otherProps}>
+      {multipartBadge}
+      <InnerStepCard className={className}>
+        {questionNumber && isHomework && stepType === 'exercise' &&
+          <StepCardHeader>
+            <div>
+              {leftHeaderChildren}
+              <div className="question-info">
+                {headerTitleChildren}
+                <span>Question {questionNumber}</span>
+                <span className="num-questions">&nbsp;/ {numberOfQuestions}</span>
+                <span className="separator">|</span>
+                <span className="exercise-id">ID: {exerciseId}</span>
+              </div>
+            </div>
+            <div>
+              <div className="points">{availablePoints} Points</div>
+              {rightHeaderChildren}
+            </div>
+          </StepCardHeader>
+        }
+        <StepCardQuestion unpadded={unpadded}>{children}</StepCardQuestion>
+      </InnerStepCard>
+    </OuterStepCard>
+  )
+};
+
+export interface TaskStepCardProps extends SharedProps {
+  className?: string;
+  children?: ReactNode;
+  step: Step;
+}
+
+const TaskStepCard = ({
+  step,
+  questionNumber,
+  numberOfQuestions,
+  children,
+  className,
+  ...otherProps
+}: TaskStepCardProps) =>
+(<StepCard {...otherProps}
+  unpadded={true}
+  questionNumber={questionNumber}
+  numberOfQuestions={numberOfQuestions}
+  stepType={step.type}
+  isHomework={step.task.type === 'homework'}
+  data-task-step-id={step.id}
+  availablePoints={step.available_points}
+  className={cn(`${step.type}-step`, className)}
+  exerciseId={step.uid}
+>
+  {children}
+</StepCard>);
+
+TaskStepCard.displayName = 'TaskStepCard';
+
+export { StepCard, TaskStepCard, LoadingCard };
```

Comment:
LoadingCard doesn't do anything, the actual loading animation component is still in Tutor, so maybe we end up removing that. And this entire file would feel better to me as a single, simpler named component.

Reply:
yeah, seems like we will need to revisit this one. my preference would be to exclude any styles that are not shared and just cross that bridge when we come to it but I'm ok with leaving it as is for now.

Reply:
Ya, not sure why I didn't remove it on Friday, went ahead and added a follow up commit to yank it.